text/0000-manageable-http-proxies.md

```diff
@@ -0,0 +1,65 @@
+- Feature Name: Manageable HTTP Proxies
+- Start Date: 2016-10-03
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]:
+
+Adds HTTP Proxies as a first class citizen that is manageable by users for use with syncing and potentially other things.
+
+# Motivation
+[motivation]:
+
+Currently when synchronizing repositories, users can either sync straight to the content source, or globally use an http proxy defined at installation time.  In many cases, you may want to use one proxy for anything syncing outside your network, but no proxy for internal sources.  There may be other cases where you want to use one proxy for syncing one repo and another proxy for a second repo.
+
+Allowing the user to define their proxies and select them on a per-repo basis would be extremely helpful in these situations.
+
+In addition it could allow easy use of proxies with capsule syncing (which is a requested feature).
+
+# Detailed design
+[design]:
+
+Most of this could be implemented in a new plugin so that other plugins could use this concept.  Alternatively it could be implemented directly in Katello.  Curious to get thoughts around this, but I could see this being useful for compute resources and the docker plugin.
```

Comment:
+1, compute resources would be a great addition. Imagine a corporate environment where your Foreman/Katello server is in an internal network and you want to talk to - let's say - the AWS API which is on the dangerous and evil internet. Your network engineers will probably ask you to use a (mitm, therefore more secure) proxy.
We might even add this to core, when it is used in some places.